dashboard/app/jobs/evaluate_rubric_job.rb

```diff
@@ -157,25 +218,36 @@ def self.get_lesson_s3_name(script_level)
     end
   end
 
-  private def write_ai_evaluations(user, ai_evaluations, rubric, channel_id, project_version)
-    _owner_id, project_id = storage_decrypt_channel_id(channel_id)
-
-    # record the ai evaluations to the database
-    # TODO: pass along and update the 'requester' to the correct id
+  private def write_ai_evaluations(user, ai_evaluations, rubric, rubric_ai_evaluation, project_version)
+    # Record the AI evaluations atomically to the database
     ActiveRecord::Base.transaction do
-      ai_evaluations.each do |evaluation|
-        learning_goal = rubric.learning_goals.all.find {|lg| lg.learning_goal == evaluation['Key Concept']}
-        understanding = understanding_s_to_i(evaluation['Grade'])
+      # Update the base rubric status
+      rubric_ai_evaluation.status = SharedConstants::RUBRIC_AI_EVALUATION_STATUS[:SUCCESS]
+
+      # Record the version of the code we evaluated
+      rubric_ai_evaluation.project_version = project_version
+
+      # Write out all of the learning goal records
+      ai_mapping = {}
+      ai_evaluations.each do |ai_evaluation|
+        ai_mapping[ai_evaluation['Key Concept']] = ai_evaluation
+      end
```

Comment:
if you like one liners, I believe this can be:
```suggestion
ai_mapping = ai_evaluations.map {|ai_eval| [ai_eval['Key Concept'], ai_eval]}.to_h
```
optional and ok if you prefer the readability of what you have here!